src/app/pages/system/update/update.component.ts

```diff
@@ -114,9 +114,8 @@ export class UpdateComponent implements OnInit {
           this.checkable = true;
           this.fullTrainList = trains.trains;
 
-          // On page load, make sure we are working with train of the current OS
-          this.train = trains.current;
-          this.selectedTrain = trains.current;
+          this.train = trains.selected;
+          this.selectedTrain = trains.selected;
```

Comment:
@RehanY147 why do we need 2 same values in 2 different variables?

Reply:
One is the value of the `ix-select` and one is to remember what the original selected train was so we don't show a warning when they switch back to that.